### PR TITLE
Zephyr small fixes

### PR DESCRIPTION
**Pull request by glima**

### CODE REVIEW

File: src/lib/common/sol-platform-impl-zephyr.c [hunk 1]
```diff
@@ -100,8 +101,13 @@ sol_platform_impl_get_serial_number(char **number)
 int
 sol_platform_impl_get_os_version(char **version)
 {
-    SOL_WRN("Not implemented");
-    return -ENOTSUP;
+    SOL_NULL_CHECK(version, -EINVAL);
+
+    *version = strdup(KERNEL_VERSION_STRING);
+    if (!*version)
+        return -ENOMEM;
```
Comment (cabelitos) [inline src/lib/common/sol-platform-impl-zephyr.c]: why not `SOL_NULL_CHECK()` ?

Comment (glima) [inline src/lib/common/sol-platform-impl-zephyr.c]: Fair (copy and pasta from RIOT's).

**Comment by glima** [comment]
@ibriano, could you take a look, please?

File: src/shared/sol-random.c [hunk 2]
```diff
@@ -91,7 +91,11 @@ get_platform_seed(uint64_t seed)
 
     /* Fall back to using a bad source of entropy if platform-specific,
      * higher quality random sources, are unavailable. */
+#ifdef SOL_PLATFORM_ZEPHYR
+    return ((uint64_t)sys_rand32_get() << 32) | sys_rand32_get();
```
Comment (ibriano) [inline src/shared/sol-random.c]: Is this always available or it depends on some driver configuration?

File: src/lib/common/sol-platform-impl-zephyr.c [hunk 3]
```diff
@@ -85,10 +85,22 @@ sol_platform_impl_set_target(const char *target)
...
 {
-    SOL_WRN("Not implemented");
+#ifdef SOL_MACHINE_ID
+    static const char ID[] = SOL_MACHINE_ID;
+    const char *ptr;
+    int i = 0;
+
+    for (ptr = ID; *ptr && i < 33; ptr++) {
+        id[i++] = *ptr;
+    }
```
Comment (ibriano) [inline src/lib/common/sol-platform-impl-zephyr.c]: SOL_MACHINE_ID is a binary string? How do you pass it as a define?

File: src/shared/sol-random.c (hunk 2 above)
Comment (glima) [inline src/shared/sol-random.c]: It *used* to be on always, recently it seems we need a config (and the config was changed wrongly, I had to patch the Kconfig) :/ But it's the recommended way of getting an almost-random number in the OS, it seems.

File: src/lib/common/sol-platform-impl-zephyr.c (hunk 3 above)
Comment (glima) [inline src/lib/common/sol-platform-impl-zephyr.c]: It's meant to get a string, yes, as in "4d77755382fb459abdd2181de3316aaa".

File: src/shared/sol-random.c (hunk 2 above)
Comment (ibriano) [inline src/shared/sol-random.c]: On Thu, 07 Apr 2016 13:49:44 -0700, Gustavo Lima Chaves wrote:
> > @@ -91,7 +91,11 @@ get_platform_seed(uint64_t seed)
> >  
> >      /* Fall back to using a bad source of entropy if platform-specific,
> >       * higher quality random sources, are unavailable. */
> > +#ifdef SOL_PLATFORM_ZEPHYR
> > +    return ((uint64_t)sys_rand32_get() << 32) | sys_rand32_get();
> 
> It *used* to be on always, recently it seems we need a config (and the config was changed wrongly, I had to patch the Kconfig) :/ But it's the recommended way of getting an almost-random number in the OS, it seems.

Ok then

File: src/lib/common/sol-platform-impl-zephyr.c (hunk 3 above)
Comment (ibriano) [inline src/lib/common/sol-platform-impl-zephyr.c]: On Thu, 07 Apr 2016 13:50:24 -0700, Gustavo Lima Chaves wrote:
> >  {
> > -    SOL_WRN("Not implemented");
> > +#ifdef SOL_MACHINE_ID
> > +    static const char ID[] = SOL_MACHINE_ID;
> > +    const char *ptr;
> > +    int i = 0;
> > +
> > +    for (ptr = ID; *ptr && i < 33; ptr++) {
> > +        id[i++] = *ptr;
> > +    }
> 
> It's meant to get a string, yes, as in "4d77755382fb459abdd2181de3316aaa".

Then why not just strcpy?

Comment (glima) [inline src/lib/common/sol-platform-impl-zephyr.c]: My original attempt was with a memcpy(), but it does not work.

**Comment by glima** [comment]
So, merging with the due fixes.

**Pull request closed without merging**